packages/api-key-stamper/src/tink/elliptic_curves.ts

```diff
@@ -136,35 +136,86 @@ function getY(x: bigint, lsb: boolean): bigint {
 }
 
 /**
- * Decodes a public key in _compressed_ format.
+ *
+ * Given x and y coordinates of a JWK, checks whether these are valid points on
+ * the P-256 elliptic curve.
+ *
+ * P-256 only
+ *
+ * @param x x-coordinate
+ * @param y y-coordinate
+ * @return boolean validity
+ */
+function validateUncompressedXY(x: bigint, y: bigint): boolean {
+  const p = getModulus();
+  const a = p - BigInt(3);
+  const b = getB();
+  const rhs = ((x * x + a) * x + b) % p;
+  const lhs = y ** BigInt(2) % p;
+  return lhs === rhs;
+}
+
+/**
+ * Decodes a public key in _compressed_ OR _uncompressed_ format.
  * Augmented to ensure that the x and y components are padded to fit 32 bytes.
  *
  * P-256 only
  */
 export function pointDecode(point: Uint8Array): JsonWebKey {
   const fieldSize = fieldSizeInBytes();
-
-  if (point.length !== 1 + fieldSize) {
-    throw new Error("compressed point has wrong length");
-  }
-  if (point[0] !== 2 && point[0] !== 3) {
-    throw new Error("invalid format");
+  const compressedLength = fieldSize + 1;
```

Comment:
This looks like a lot of new code, and we're in a **vendored** file! At this point it'd be really hard for someone to understand how to upgrade tink if we ever had to.

Still, can you add a summary of what was done in this PR at the top of the file?

Reply:
Ah yeah wanted to follow the earlier suggestion of moving the uncompressed Pub key point into pointDecode -- didn't realize this is a vendored file until later will definitely add some context on top.

Also -- it seems like we've changed the file significantly from what the original implementation is for our own purposes. The original implementation also does have the capacity to check uncompressed points!

Reply:
(all this is to say that our implementation is already quite different from what's in the original, and this actually kinda brings it closer to the original)